src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncSpillTests.cs

```diff
@@ -3888,5 +3887,2493 @@ class AltBoolean
+    IL_0105:  ldloca.s   V_2
+    IL_0107:  ldarg.0
+    IL_0108:  call       ""void System.Runtime.CompilerServices.AsyncTaskMethodBuilder.AwaitUnsafeOnCompleted<System.Runtime.CompilerServices.TaskAwaiter<int>, Program.<Main>d__0>(ref System.Runtime.CompilerServices.TaskAwaiter<int>, ref Program.<Main>d__0)""
+    IL_010d:  leave.s    IL_0180
+    IL_010f:  ldarg.0
+    IL_0110:  ldfld      ""System.Runtime.CompilerServices.TaskAwaiter<int> Program.<Main>d__0.<>u__1""
+    IL_0115:  stloc.2
+    IL_0116:  ldarg.0
+    IL_0117:  ldflda     ""System.Runtime.CompilerServices.TaskAwaiter<int> Program.<Main>d__0.<>u__1""
+    IL_011c:  initobj    ""System.Runtime.CompilerServices.TaskAwaiter<int>""
+    IL_0122:  ldarg.0
+    IL_0123:  ldc.i4.m1
+    IL_0124:  dup
+    IL_0125:  stloc.0
+    IL_0126:  stfld      ""int Program.<Main>d__0.<>1__state""
+    IL_012b:  ldloca.s   V_2
+    IL_012d:  call       ""int System.Runtime.CompilerServices.TaskAwaiter<int>.GetResult()""
+    IL_0132:  stloc.1
+    IL_0133:  ldarg.0
+    IL_0134:  ldfld      ""B Program.<Main>d__0.<>7__wrap3""
+    IL_0139:  ldloc.1
+    IL_013a:  stfld      ""int B.x""
+    IL_013f:  ldarg.0
+    IL_0140:  ldnull
+    IL_0141:  stfld      ""B Program.<Main>d__0.<>7__wrap3""
+    IL_0146:  leave.s    IL_0166
+  }
+  catch System.Exception
+  {
+    IL_0148:  stloc.3
+    IL_0149:  ldarg.0
+    IL_014a:  ldc.i4.s   -2
+    IL_014c:  stfld      ""int Program.<Main>d__0.<>1__state""
+    IL_0151:  ldarg.0
+    IL_0152:  ldnull
+    IL_0153:  stfld      ""A Program.<Main>d__0.<a>5__2""
+    IL_0158:  ldarg.0
+    IL_0159:  ldflda     ""System.Runtime.CompilerServices.AsyncTaskMethodBuilder Program.<Main>d__0.<>t__builder""
+    IL_015e:  ldloc.3
+    IL_015f:  call       ""void System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetException(System.Exception)""
+    IL_0164:  leave.s    IL_0180
+  }
+  IL_0166:  ldarg.0
+  IL_0167:  ldc.i4.s   -2
+  IL_0169:  stfld      ""int Program.<Main>d__0.<>1__state""
+  IL_016e:  ldarg.0
+  IL_016f:  ldnull
+  IL_0170:  stfld      ""A Program.<Main>d__0.<a>5__2""
+  IL_0175:  ldarg.0
+  IL_0176:  ldflda     ""System.Runtime.CompilerServices.AsyncTaskMethodBuilder Program.<Main>d__0.<>t__builder""
+  IL_017b:  call       ""void System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetResult()""
+  IL_0180:  ret
+}");
+        }
+
+        [Fact]
+        [WorkItem(42755, "https://github.com/dotnet/roslyn/issues/42755")]
+        public void KeepLtrSemantics_ClassFieldAccessOnArray()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+
+class Program
+{
+    static async Task Main(string[] args)
+    {
+        var a = new A[1];
+        try
+        {
+            a[1].x = await Write(""1"");
+        }
+        catch 
+        {
+            try
+            {
+                a[0].x = await WriteAndSet(""0"");
```

Comment:
>a[0].x = await WriteAndSet(""0""); [](start = 16, length = 34)

It looks like we a bundling a lot into the same unit-test. Because of that the IL is huge and it is not trivial to verify the expectations. Consider breaking up the tests to make them more granular and focused on a specific aspects. For example, it is not clear whether having this assignment inside a ```try``` nested into a ```catch``` is really motivated by a goal to test something specific for that nested ```try/catch``` structure. Assuming that it is not, and ```catch``` is there just to catch an exception We could have one helper function with doing something like this:
```
        try
        {
            <assignment here>;
        }
        catch 
        {
             <print exception info>
        }                       
``` 

The caller can verify what side-effects happened during the try, if any are expected to happen.

The input to that helper (perhaps more than one parameter) can drive
- what and whether an exception is thrown and at what time,
- what side-effects evaluation of the right hand side of the assignment is supposed to have,
- etc 

This way, with one small function we will be able to easily test multiple execution scenarios, yet minimize the IL to verify.

Reply:
It looks like this suggestion is applicable to pretty much all the new tests.

---
In reply to: [400571262](https://github.com/dotnet/roslyn/pull/42807#discussion_r400571262) [](ancestors = 400571262)

Reply:
Extracting the assignment into its own method would be even more useful for the purpose of IL verification.

---
In reply to: [400571923](https://github.com/dotnet/roslyn/pull/42807#discussion_r400571923) [](ancestors = 400571923,400571262)

Reply:
@YairHalberstadt I didn't see a response to this comment.

Reply:
@333fred I'm not sure what you mean

Reply:
@YairHalberstadt specifically this part:
> Consider breaking up the tests to make them more granular and focused on a specific aspects.

I didn't see any test changes to adapt to the pattern Aleksey suggested.

Reply:
That being said, I think my vscode instance was messing up because I see changes online, but didn't see them locally. I'll take another look now.